registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/SyncRegistrationServiceImpl.java

```diff
@@ -575,6 +615,46 @@ public RegistrationSyncRequestDTO decryptAndGetSyncRequest(Object encryptedSyncM
 
 	}
 
+	@Override
+	public LostRidDto searchLostRid(SearchInfo searchInfo) {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"SyncRegistrationServiceImpl::getByIds()::entry");
+		try {
+			LostRidDto lostRidDto = new LostRidDto();
+			createSearchInfoDto(searchInfo);
+			List<String> registrationIds = syncRegistrationDao.getSearchResults(searchInfo.getFilters(),
+					searchInfo.getSort());
+			validateRegistrationIds(registrationIds);
+			lostRidDto.setRegistartionIds(registrationIds);
+			return lostRidDto;
+		} catch (DataAccessLayerException | NoSuchAlgorithmException | RegStatusAppException e) {
+
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", e.getMessage() + ExceptionUtils.getStackTrace(e));
+			throw new TablenotAccessibleException(
+					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getMessage(), e);
+		}
+	}
+
+	private void validateRegistrationIds(List<String> registrationIds) throws RegStatusAppException {
+		LostRidValidationException exception = new LostRidValidationException();
+		if (registrationIds.size() >= maxSearchResult) {
+			throw new RegStatusAppException(PlatformErrorMessages.RPR_RGS_INVALID_SEARCH, exception);
+		}
+
+	}
+
+	private void createSearchInfoDto(SearchInfo searchInfo) throws NoSuchAlgorithmException {

```

Comment:
Rename this method updateFiltersWithHashedValues

Reply:
taken care